### PR TITLE
Logically separate secrets and keys

### PR DESCRIPTION
**Pull request by jcox-dev**
- Renamed key files to follow sdc-crypto convention of `<platform>-<service>-<purpose>-<key_use>-<key_type>-<version>.pem`
- Used sdc-crypto functionality to generate key files.
- Separated the generating of keys and secrets into individual scripts and yaml files.
- Updated the initialisation of the SecretStore to expect individual objects for secrets and keys (after validation).
- Secrets now found (after being generated) in secrets.yml and keys now held in keys.yml.

### What is the context of this PR?
**As a** service manager
**I want** to separate secrets from keys
**So that** we follow best practices

### How to review 
Are files added or renamed now consistent with the rest of the repo?
Are secrets and keys logically separated?
Do all the tests pass?
Do all the deployment scripts still work as expected? (Filename changes may have caused some issues for prod and pre-prod).

### CODE REVIEW

File: app/secrets.py [hunk 1]
```diff
@@ -22,15 +22,12 @@ def validate_required_secrets(secrets):
         if required_secret not in secrets['secrets']:
             raise Exception("Missing Secret [{}]".format(required_secret))
 
-    validate_required_submission_keys(secrets)
 
-
-def validate_required_submission_keys(secrets):
+def validate_required_submission_keys(keys):
```
Comment (jonnyshaw89) [inline app/secrets.py]: I think you should be able to use this method from the library.
https://github.com/ONSdigital/sdc-cryptography/blob/master/sdc/crypto/key_store.py#L10

File: app/secrets.py [hunk 2]
```diff
@@ -63,14 +60,12 @@ def as_jwk(self):
...
-        self.secrets = secrets.get('secrets')
-        if 'keys' in secrets:
-            self.keys = {}
-            for _, kid in enumerate(secrets['keys']):
-                key = secrets['keys'][kid]
-                key_object = Key(kid, key['purpose'], key['type'], key['value'])
-                self.keys[kid] = key_object
+    def __init__(self, keys, secrets=None):
+        self.secrets = secrets.get('secrets') if secrets is not None else {}
+        self.keys = {}
```
Comment (jonnyshaw89) [inline app/secrets.py]: we should remove keys entirely from he secret store and have a KeyStore https://github.com/ONSdigital/sdc-cryptography/blob/master/sdc/crypto/key_store.py#L44

File: app/setup.py [hunk 3]
```diff
@@ -60,8 +60,10 @@ def create_app(setting_overrides=None):  # noqa: C901  pylint: disable=too-compl
     application.eq = {}
 
     secrets = yaml.safe_load(open(application.config['EQ_SECRETS_FILE']))
+    keys = yaml.safe_load(open(application.config['EQ_KEYS_FILE']))
     validate_required_secrets(secrets)
-    application.eq['secret_store'] = SecretStore(secrets)
+    validate_required_submission_keys(keys)
+    application.eq['secret_store'] = SecretStore(keys, secrets)
```
Comment (jonnyshaw89) [inline app/setup.py]: as mentioned above we should have a `secret_store` and a `key_store`

**Comment by shahi645** [comment]
I get some linting errors in *secrets.py* and i think the eq keys names should change from *sr* to *eq*. Otherwise this PR looks good to me.

File: jenkins-preprod.sh [hunk 4]
```diff
@@ -4,3 +4,4 @@ echo  $Password | gpg -o secure-data.tar --passphrase-fd 0 --batch /opt/keys/sec
 tar xvf secure-data.tar
 
 python scripts/generate_secrets.py secure-data/
+python scripts/generate_keys.py secure-data/
```
Comment (jonnyshaw89) [inline jenkins-preprod.sh]: I think we should call this script and remove our one
https://github.com/ONSdigital/sdc-cryptography/blob/master/sdc/crypto/scripts/generate_keys.py

File: scripts/generate_keys.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
```
Comment (jonnyshaw89) [inline scripts/generate_keys.py]: We shouldn't need this.

File: tests/integration/introduction/test_introduction.py [hunk 6]
```diff
@@ -9,7 +9,7 @@ def test_mail_link_contains_ru_ref_in_subject(self):
 
         # When on the landing page
         # Then the email link is present with the ru_ref in the subject
-        self.assertRegexPage('\"mailto\:.+\?subject\=.+123456789012A\"')
+        self.assertRegexPage(r'\"mailto\:.+\?subject\=.+123456789012A\"')
```
Comment (jonnyshaw89) [inline tests/integration/introduction/test_introduction.py:12]: Is this change related to this PR?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: jenkins-preprod.sh (hunk 4 above)
Comment (shahi645) [reply jenkins-preprod.sh]: I think theres issues with the *kid_override* in the SR code, but we could submit a PR to the sdc-crypo repo

File: tests/integration/introduction/test_introduction.py (hunk 6 above)
Comment (jcox-dev) [reply tests/integration/introduction/test_introduction.py:12]: Linting error blocked build. Could be committed elsewhere, sure.

**Comment by jcox-dev** [comment]
@shahi645 @jonnyshaw89 changes pushed, separating the KeyStore and SecretStore.

File: jenkins-preprod.sh (hunk 4 above)
Comment (jonnyshaw89) [reply jenkins-preprod.sh]: @shahi645 this isn't an issue anymore as RRM have deployed a change so we no longer require these overrides

**Review by shahi645 — APPROVED** [review]
Keys are generated using sdc-crypto. All works as should!

File: app/secrets.py [hunk 7]
```diff
@@ -1,6 +1,6 @@
-from jwcrypto import jwk
+KEY_PURPOSE_AUTHENTICATION = 'authentication'
```
Comment (jonnyshaw89) [inline app/secrets.py]: Is this file still the correct place for these?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]
You will also need to change `docker-compose.yml` and `docker-secrets.yml`

File: docker-secrets.yml [hunk 8]
```diff
@@ -131,12 +131,3 @@ keys:
       V7ZuRp//4vIzEONqqyLUFt1JZKuCwWQB2sRjKWpqRB07C0o243XefD5f7844nmXq
       H0xKMFRYjDqhFdTXFWVPnk/tu/MxoSTdK0FfsgUnjs+TGvN5TnBW
       -----END RSA PRIVATE KEY-----
-secrets:
```
Comment (jonnyshaw89) [inline docker-secrets.yml]: You should also remove `EDCRRM` and `EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY` from this file

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: docker-keys.yml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+secrets:
```
Comment (jonnyshaw89) [inline docker-keys.yml]: Also these files are the wrong way around.
`docker-keys.yml` should contain keys not secrets and visa versa

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

**Review by jonnyshaw89 — APPROVED** [review]

**Comment by jonnyshaw89** [comment]
@jcox-dev can you squash your commits?
